fhir-path/src/main/java/com/ibm/fhir/path/function/ResolveFunction.java

```diff
@@ -94,16 +112,49 @@ public int getMaxArity() {
                             }
                         }
                     } else {
-                        Matcher matcher = REFERENCE_PATTERN.matcher(referenceReference);
-                        if (matcher.matches()) {
-                            resourceType = matcher.group(RESOURCE_TYPE_GROUP);
-                            if (referenceType != null && !resourceType.equals(referenceType)) {
-                                throw new IllegalArgumentException("Resource type found in reference URL does not match reference type");
+                        if (isBundleContext) {
+                            // we know the root of the tree is a Bundle and the current node is of type Reference,
+                            // so walk up the tree until we get to highest Bundle.entry.fullUrl in the tree
+                            // and save the value of its fullUrl
+                            String fullUrl = null;
+                            FHIRPathTree tree = evaluationContext.getTree();
+                            FHIRPathNode nodeUnderTest = tree.getParent(node);
+                            while (nodeUnderTest != root) {
+                                if (nodeUnderTest.isResourceNode()) {
+                                    FHIRPathNode sibling = tree.getSibling(nodeUnderTest, "fullUrl");
+                                    fullUrl = sibling.getValue().asStringValue().string();
```

Comment:
Looks like null checks are done a lot, except here.  Just making sure this won't ever be null (or check for it, if not).

Reply:
honestly not sure if its possible for it to be null, but won't hurt to add some guards...will update

Reply:
done